### PR TITLE
Fixes mail changing existing nuke codes

### PR DESCRIPTION
**Pull request by GoblinBackwards**
## About The Pull Request
Fixes an issue with the mail which reveals part of the nuke code that meant it could change the code for the station nuke even it one had already been generated.

## Why It's Good For The Game
bug moment

## Changelog

:cl:
fix: Fixed mail being able to change the stations nuke code when it had already been generated.
/:cl:

### CODE REVIEW

File: code/game/objects/items/mail.dm [hunk 1]
```diff
@@ -300,14 +300,15 @@
...
 		return ..()
-	var/code = random_nukecode()
-	for(var/obj/machinery/nuclearbomb/selfdestruct/self_destruct in GLOB.nuke_list)
-		self_destruct.r_code = code
-	message_admins("Through junkmail, the self-destruct code was set to \"[code]\".")
-	add_raw_text("<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[code[rand(1,5)]][code[rand(1,5)]]...'")
+
+	if(self_destruct.r_code == "ADMIN")
+		self_destruct.r_code = random_nukecode()
+	message_admins("Through junkmail, the self-destruct code was set to \"[self_destruct.r_code]\".")
```
Comment (optimumtact) [inline code/game/objects/items/mail.dm:310]: this needs to be indented

File: code/game/objects/items/mail.dm [hunk 2]
```diff
@@ -300,14 +300,15 @@
...
+	if(!self_destruct || !prob(nuclear_option_odds)) // 1 in 1000 chance of getting 2 random nuke code characters.
 		add_raw_text("<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[rand(0,9)][rand(0,9)][rand(0,9)]...'")
 		return ..()
-	var/code = random_nukecode()
-	for(var/obj/machinery/nuclearbomb/selfdestruct/self_destruct in GLOB.nuke_list)
-		self_destruct.r_code = code
-	message_admins("Through junkmail, the self-destruct code was set to \"[code]\".")
-	add_raw_text("<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[code[rand(1,5)]][code[rand(1,5)]]...'")
+
+	if(self_destruct.r_code == "ADMIN")
```
Comment (Mothblocks) [inline code/game/objects/items/mail.dm:308]: For this and everywhere that uses "ADMIN", change it to a define somewhere in __DEFINES.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__DEFINES/nuclear_bomb.dm [hunk 3]
```diff
@@ -26,3 +26,5 @@
 #define DETONATION_HIT_SYNDIE_BASE NUKE_SYNDICATE_BASE
 #define DETONATION_NEAR_MISSED_STATION NUKE_NEAR_MISS
 #define DETONATION_MISSED_STATION NUKE_MISS_STATION
+
+#define DEFAULT_NUKE_CODE "ADMIN"
```
Comment (Mothblocks) [inline code/__DEFINES/nuclear_bomb.dm:30]: 1. Can you rename it to something like `NUKE_CODE_UNSET` or something to be clear that this is an invalid code meant specifically for this purpose?
2. Auto doc it (triple slashes above the define)

**Review by Mothblocks — CHANGES_REQUESTED** [review]
One last nit

File: code/game/objects/items/mail.dm (hunk 2 above)
Comment (Mothblocks) [inline code/game/objects/items/mail.dm:308]:
```suggestion
	if(self_destruct.r_code == NUKE_CODE_UNSET)
```

**Review by Mothblocks — APPROVED** [review]

**Comment by timothymtorres** [comment]
Interesting.  I wonder if this is what was causing #70419

**Comment by Mothblocks** [comment]
@timothymtorres In the round given, the random_nukecode proc is only called once. Not likely.

**Comment by Mothblocks** [comment]
profiler.json

**Pull request merged**